### PR TITLE
Closes #2611 - CPCSS for mobile: tools part (2/4)

### LINKED ISSUE #2611: CPCSS for mobile: tools part (2/4)
Sub-task for epic #2378 

### views/settings/page-sections/tools.php
- [x] Add `do_action( 'rocket_settings_tools_content' );` to the view

### views/cpcss
- [x] Add a `activate-cpcss-mobile.php` view file, containing a description + button to enable the CPCSS for mobile

### inc/Engine/CriticalPath/AdminSubscriber.php
- [x] Add `display_cpcss_mobile_section()` method as a callback for `rocket_settings_tools_content` to display the "active CPCSS for mobile" view when needed (Optimize CSS delivery + specific cache for mobile options enabled)
- [x] Add `enable_cpcss_mobile()` method AJAX callback to enable the `async_css_mobile` option via the button in the view
- [x] Add tests

### PR DESCRIPTION
**Pull request by crystinutzaa**
Sub-task for epic #2378 

### views/settings/page-sections/tools.php
- [x] Add `do_action( 'rocket_settings_tools_content' );` to the view

### views/cpcss
- [x] Add a `activate-cpcss-mobile.php` view file, containing a description + button to enable the CPCSS for mobile

### inc/Engine/CriticalPath/AdminSubscriber.php
- [x] Add `display_cpcss_mobile_section()` method as a callback for `rocket_settings_tools_content` to display the "active CPCSS for mobile" view when needed (Optimize CSS delivery + specific cache for mobile options enabled)
- [x] Add `enable_cpcss_mobile()` method AJAX callback to enable the `async_css_mobile` option via the button in the view
- [x] Add tests

TODO:
- [x]  Fix current Unit tests
- [x]  Fix current Integration tests

Add tests:
- [x] - Unit Test for `display_cpcss_mobile_section()` - `Ahmed`
- [x] - Integration test for `display_cpcss_mobile_section()` - `Ahmed`
- [x] - Unit Test for `enable_mobile_cpcss()` - `Cristina`
- [x] - Integration test for `enable_mobile_cpcss()` - `Cristina`

- [x] - QA

### CODE REVIEW

File: views/settings/page-sections/tools.php [hunk 1]
```diff
@@ -87,6 +87,8 @@
 		</div>
 	</div>
 
+	<?php do_action( 'rocket_settings_tools_content' ); ?>
```
Comment (Tabrisrp) [inline views/settings/page-sections/tools.php:90]: add DocBlock for the action

**Comment by crystinutzaa** [comment]
@Tabrisrp , @Tabrisrp this one is blocked by the merge:

**Comment by crystinutzaa** [comment]
@hellofromtonya , @Tabrisrp this one is blocked by this merge due to the Unit test which needs the `generate()`
`There are too many changes in develop and feature/cpcss-for-specific-pages.`

In rest it seems ready. So if you have some time your early review would be helpful.

**Comment by Tabrisrp** [comment]
Code looks good up to now, we're still waiting on product to settle on the final implementation though

**Comment by crystinutzaa** [comment]
@wp-media/dev-qa this one needs testing and brainstorming on the UI and that hide on click action.

**Comment by arunbasillal** [comment]
QA:

- UI is as per spec 👍 
image

- This is not displayed if Optimize CSS delivery is disabled 👍 
- Not displayed if Separate cache for mobile is disabled 👍 

- Before clicking button
image
(Expected since I am switching branches on GitHub) 👍 

- Clicking the button didn't do anything 🤔 - https://jmp.sh/mBCmjY3

@crystinutzaa @engahmeds3ed @Tabrisrp Isn't this PR supposed to set `async_css_mobile` to `1`?

**Comment by Tabrisrp** [comment]
Yes it is, that's what the button is doing. I just tried on my local and it worked without issues?

**Comment by arunbasillal** [comment]
Continued QA: 

- On new install `async_css_mobile` was set to `1` 👍 
- Still didn't work on my old test site. No idea why, no other plugins were active. 🤔 
- The UI works on a different local install - https://youtu.be/XV05KLfiG2s 👍 
- But it didn't on the mega site - https://youtu.be/NHbpK86BO0M (installed here: http://mega.wp-rocket.me/wp-admin/options-general.php?page=wprocket#tools)

On the install that it worked, the UX was poor imo. @GeekPress @webtrainingwheels can you please take a look here: https://youtu.be/XV05KLfiG2s 
- I clicked the button at the beginning of the video. There was no feedback
- When I refreshed the page, the button was gone. 
- Is this the intended design?

**Comment by webtrainingwheels** [comment]
@arunbasillal Agree, that looks like a weird experience.  I feel like we should remove the button but add a little confirmation text. 
> Your site is now using mobile-specific critical path CSS.

**Comment by crystinutzaa** [comment]
@arunbasillal please re-test this

**Comment by arunbasillal** [comment]
@crystinutzaa It worked as per our discussion on `Mega/Divi`. 

However clicking the button did nothing on my Local and the main Mega site - https://jmp.sh/ToeiYT8
I had the same issue yesterday as well. Please take a look here: http://mega.wp-rocket.me/wp-admin/options-general.php?page=wprocket#tools

**Pull request merged**